botocore/credentials.py

```diff
@@ -122,6 +124,46 @@ def create_credential_resolver(session, cache=None):
     return resolver
 
 
+class ProfileProviderBuilder(object):
+
+    def __init__(self, session, cache=None):
+        self._session = session
+        self._cache = cache
+        self._profile = (
+            self._session.get_config_variable('profile') or 'default'
+        )
+
+    def providers(self, profile_name=None):
```

Comment:
We should also add general unit tests for this class even if we are marking it as internal.